plugins/ipam/host-local/allocator_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"github.com/containernetworking/cni/pkg/types"
+	fakestore "github.com/containernetworking/cni/plugins/ipam/host-local/backend/testing"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"net"
+)
+
+var _ = Describe("host-local allocator", func() {
+	It("should allocate ips in round robin", func() {
+		testCases := []struct {
+			desc         string
+			subnet       string
+			ipmap        map[string]string
+			expectErr    bool
+			expectResult string
+			lastIP       string
+		}{
+			// fresh start
+			{
+				subnet:       "10.0.0.0/29",
+				ipmap:        map[string]string{},
+				expectErr:    false,
+				expectResult: "10.0.0.2",
+				lastIP:       "",
+			},
+			{
+				subnet: "10.0.0.0/29",
+				ipmap: map[string]string{
+					"10.0.0.2": "id",
+				},
+				expectErr:    false,
+				expectResult: "10.0.0.3",
+				lastIP:       "",
+			},
+			// next ip of last reserved ip
+			{
+				subnet:       "10.0.0.0/29",
+				ipmap:        map[string]string{},
+				expectErr:    false,
+				expectResult: "10.0.0.7",
+				lastIP:       "10.0.0.6",
+			},
+			{
+				subnet: "10.0.0.0/29",
+				ipmap: map[string]string{
+					"10.0.0.5": "id",
+					"10.0.0.6": "id",
+				},
+				expectErr:    false,
+				expectResult: "10.0.0.7",
+				lastIP:       "10.0.0.4",
+			},
+			// round robin to the beginning
+			{
+				subnet: "10.0.0.0/29",
+				ipmap: map[string]string{
+					"10.0.0.7": "id",
+				},
+				expectErr:    false,
+				expectResult: "10.0.0.2",
+				lastIP:       "10.0.0.6",
+			},
+			// lastIP is out of range
+			{
+				subnet: "10.0.0.0/29",
+				ipmap: map[string]string{
+					"10.0.0.2": "id",
+				},
+				expectErr:    false,
+				expectResult: "10.0.0.3",
+				lastIP:       "10.0.0.128",
+			},
+			// out of ips
+			{
+				subnet: "10.0.0.0/30",
+				ipmap: map[string]string{
+					"10.0.0.2": "id",
+					"10.0.0.3": "id",
+				},
+				expectErr: true,
+			},
+		}
+
+		for _, tc := range testCases {
+			subnet, err := types.ParseCIDR(tc.subnet)
+			conf := IPAMConfig{
+				Name:   "test",
+				Type:   "host-local",
+				Subnet: types.IPNet{IP: subnet.IP, Mask: subnet.Mask},
+			}
+			store := fakestore.NewFakeStore(tc.ipmap, net.ParseIP(tc.lastIP))
+			alloc, _ := NewIPAllocator(&conf, store)
+			res, err := alloc.Get("ID")
+			if tc.expectErr {
```

Comment:
In canonical BDD style, we don't have assertions inside `if` blocks, but instead separate the different conditions into separate `Context` blocks, e.g.
```
Context("when out of ips", func() {
   BeforeEach(func() {
     // set up the conditions for tests
   })
   It("returns a meaningful error", func() {
      // run the code under test
      Expect(err).To(MatchError("no more ips"))
   })
})
```